### PR TITLE
Restore CloudSdk.validate()

### LINKED ISSUE #118: Restore CloudSdk#validate() to check a location
`CloudSdk` had a `validate()` method to check whether the location seemed like a real installation.  This check is useful for handling user-configured locations.

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes #118.

@loosebazooka @briandealwis PTAL

### CODE REVIEW

**Comment by loosebazooka** [comment]
where did it go?

**Comment by meltsufin** [comment]
I removed it previously, since the comment on it said "no one should be using it". Sorry :-(

**Comment by briandealwis** [comment]
So the only issue I have is that the _reason_ strings passed into `AppEngineException` aren't great.  It's useful to report the problems found so those who encounter a problem  (e.g., a new version of gcloud is released that changes the organization) can patch it up sufficiently to pass.

**Comment by meltsufin** [comment]
You can get the message from the exception. Is that not sufficient for your use-case?

**Comment by briandealwis** [comment]
They're not translatable, but they're enough to report on logging.  I took a quick stab at making them a bit more reportable and it's difficult (_expected 'bin/cloud.jar' to be a file_). So LGTM.

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java [hunk 1]
```diff
@@ -193,6 +193,38 @@ private Path getJavaToolsJar() {
...
+  /**
+   * Checks whether the configured Cloud SDK Path is valid.
+   *
+   * @throws AppEngineException when there is a validation error.
+   */
+  public void validate() throws AppEngineException {
+    if (sdkPath == null) {
+      throw new AppEngineException("Validation Error: SDK path is null");
+    }
+    if (sdkPath.toFile().isDirectory()) {
```
Comment (briandealwis) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: These test conditions are inverted: missing a '!'

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: :scream: how did this slip in! great catch!

**Comment by briandealwis** [comment]
I put two fixes in https://github.com/briandealwis/app-tools-lib-for-java/tree/cloudsdk_validate

**Comment by meltsufin** [comment]
@briandealwis Integrated your fixes. PTAL

**Comment by briandealwis** [comment]
LGTM

**Pull request merged**